### PR TITLE
Expose low level API to implement workflows asynchronously

### PR DESCRIPTION
**Pull request by mfateev**
Go SDK can be used as a library for implementing workflows in other languages. But the existing high level Go API is not suitable for such integrations. This PR exposes the low level async API appropriate for such integrations.

### CODE REVIEW

File: test/bindings_workflows_test.go [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package test_test
+
+import (
+	"context"
+	commonpb "go.temporal.io/temporal-proto/common"
+	"go.temporal.io/temporal/encoded"
+	b "go.temporal.io/temporal/internalbindings"
```
Comment (alexshtin) [inline test/bindings_workflows_test.go:31]: Can you alias it to `bindings`. `b` is like a reciever var.

File: test/bindings_workflows_test.go [hunk 2]
```diff
@@ -0,0 +1,140 @@
+// The MIT License
```
Comment (alexshtin) [inline test/bindings_workflows_test.go:1]: This file doesn't have any tests. It is more like `workflow_definition_factory.go`.

File: test/activity_test.go [hunk 3]
```diff
@@ -22,7 +22,7 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
 
-package test
+package test_test
```
[the remark this answers is not in the document]
Comment (mfateev) [reply test/activity_test.go:25]: I believe it makes it private to anyone who includes the package.

File: test/bindings_workflows_test.go (hunk 2 above)
Comment (mfateev) [reply test/bindings_workflows_test.go:1]: removing _test prefix makes it public :(

**Pull request merged**